components/PriceDiscountInput/index.js

```diff
@@ -0,0 +1,54 @@
+import { useState } from "react";
+import { Checkbox } from "../Checkbox";
+import { FormErrorText } from "../FormErrorText";
+import { Input } from "../Input";
+
+export function PriceDiscountInput({
+  initialPrice,
+  initialDiscount,
+  isDiscounted,
+  errors,
+  priceOption = null,
+}) {
+  const [showDiscount, setShowDiscount] = useState(isDiscounted);
+
+  function showDiscountInput(isChecked) {
+    setShowDiscount(isChecked);
+  }
+
+  const errorSource = priceOption ? errors?.priceOption : errors;
+
+  return (
+    <>
+      <label htmlFor={`price${priceOption}`}>Original price</label>
+      <Input
+        type="number"
+        name={`price__${priceOption}`}
+        id={`price__${priceOption}`}
+        defaultValue={initialPrice}
+        $error={errorSource?.price ? true : false}
+      />
+      <FormErrorText>{errorSource?.price?.message}</FormErrorText>
+      <Checkbox
+        checkboxName={`isDiscount__${priceOption}`}
```

Comment:
```suggestion
        checkboxName={`isDiscounted__${priceOption}`}
```
There's also probably more which is not part of this PR (and I can't check ...)